shell/platform/embedder/embedder.cc

```diff
@@ -910,6 +913,53 @@ FlutterEngineResult FlutterEngineInitialize(size_t version,
     };
   }
 
+  flutter::PlatformViewEmbedder::ComputePlatformResolvedLocaleCallback
+      compute_platform_resolved_locale_callback = nullptr;
+  if (SAFE_ACCESS(args, compute_platform_resolved_locale_callback, nullptr) !=
+      nullptr) {
+    compute_platform_resolved_locale_callback =
+        [ptr = args->compute_platform_resolved_locale_callback](
+            const std::vector<std::string>& supported_locales_data) {
+          size_t number_of_strings_per_locale = 3;
+          size_t locale_count =
+              supported_locales_data.size() / number_of_strings_per_locale;
+          const FlutterLocale* supported_locales[locale_count];
```

Comment:
Create a `std::vector<FlutterLocale>` and then build a `std::vector<FlutterLocale*>` containing pointers to the structs in the first vector.  This eliminates the overhead of heap allocating each `FlutterLocale` and the need to manually delete the `FlutterLocales`.